homework-submission/main.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+function AddingForm({ addTodoItem }) {
+  return (
+    <div className='form'>
+      <form onSubmit={addTodoItem} className='Add'>
+        <input type='text' name='description' placeholder='Description' className='AddingList' />
+        <input type='date' name='deadline' placeholder='Deadline' />
+        <input type='submit' name='add' value='Add' className='submit' />
+      </form>
+    </div>
+  );
+}
+
+class TodoList extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      dynamicList: [
+        {
+          id: 1,
+          description: 'Get out of bed',
+          deadline: '2017-09-11',
+          done: true,
+        },
+        {
+          id: 2,
+          description: 'Brush teeth',
+          deadline: '2017-09-10',
+          done: false,
+        },
+        {
+          id: 3,
+          description: 'Eat breakfast',
+          deadline: '2017-09-09',
+          done: false,
+        },
+      ],
+    };
+  }
+
+  addItem = element => {
+    element.preventDefault();
+    const newTodo = {
+      id: this.state.dynamicList.length + 1,
+      description: element.target.description.value,
+      deadline: element.target.deadline.value,
+      done: false,
+    };
+    console.log(newTodo);
+    this.setState({ dynamicList: this.state.dynamicList.concat(newTodo) });
+  };
+  removeItem = index => {
+    const newTodoList = this.state.dynamicList.splice(index, 1);
```

Comment:
if you tried the following you will notice a direct change of the state
```javascript
    console.log(this.state.dynamicList.length) // if this 3
    const newTodoList = this.state.dynamicList.splice(index, 1); // you are splicing here!
    console.log(this.state.dynamicList.length) // this is 2 and it should stay 3
```

This direct change shouldn't be happening, the main reason of `.setState` is making things asynchronous.

the `.setState` method isn't only for redering the content, it's made on top of a complete life cycle happening on behalf of your program, and we want to make sure we are not changing the state by hand as here.

the best solution here would be either to make a deep copy of the array, or making a new array, from js2 we remember having `.filter` will return a new array, and this is the best example of it, and where it should go.